Travis is suppressing validator test errors for PRs

From #11199, Travis is reporting that the validator test exits status 1 but it doesn't display why. When I test this locally it does state why.

Here's what Travis is displaying:
```
[13:33:06] Starting 'validator'...
npm WARN deprecated minimatch@0.3.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated minimatch@0.3.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
Traceback (most recent call last):
  File "build.py", line 580, in <module>
    Main()
  File "build.py", line 576, in Main
    RunTests(out_dir='dist', nodejs_cmd=nodejs_cmd)
  File "build.py", line 548, in RunTests
    subprocess.check_call([nodejs_cmd, '%s/test_runner' % out_dir])
  File "/opt/python/2.7.13/lib/python2.7/subprocess.py", line 186, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['node', 'dist/test_runner']' returned non-zero exit status 1
[13:35:23] 'validator' errored after 2.27 min
[13:35:23] Error: Command failed: cd validator && python build.py
npm WARN deprecated minimatch@0.3.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated minimatch@0.3.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
Traceback (most recent call last):
  File "build.py", line 580, in <module>
    Main()
  File "build.py", line 576, in Main
    RunTests(out_dir='dist', nodejs_cmd=nodejs_cmd)
  File "build.py", line 548, in RunTests
    subprocess.check_call([nodejs_cmd, '%s/test_runner' % out_dir])
  File "/opt/python/2.7.13/lib/python2.7/subprocess.py", line 186, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['node', 'dist/test_runner']' returned non-zero exit status 1
    at checkExecSyncError (child_process.js:468:13)
    at execSync (child_process.js:508:13)
    at Gulp.validator (/home/travis/build/ampproject/amphtml/build-system/tasks/validator.js:26:3)
    at module.exports (/home/travis/build/ampproject/amphtml/node_modules/orchestrator/lib/runTask.js:34:7)
    at Gulp.Orchestrator._runTask (/home/travis/build/ampproject/amphtml/node_modules/orchestrator/index.js:273:3)
    at Gulp.Orchestrator._runStep (/home/travis/build/ampproject/amphtml/node_modules/orchestrator/index.js:214:10)
    at Gulp.Orchestrator.start (/home/travis/build/ampproject/amphtml/node_modules/orchestrator/index.js:134:8)
    at /home/travis/build/ampproject/amphtml/node_modules/gulp/bin/gulp.js:129:20
    at nextTickCallbackWith0Args (node.js:489:9)
    at process._tickCallback (node.js:418:13)
    at Function.Module.runMain (module.js:443:11)
    at startup (node.js:140:18)
    at node.js:1043:3
Command failed: node_modules/gulp/bin/gulp.js validator
```

For example, here is what the output should look like (with minor differences cause it's on Travis):

```
[[build.py RunTests]] - entering ...
Started
.....F....................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
Failures:
1) ValidatorFeatures feature_tests/ads.html
  Message:
    AssertionError: 
    feature_tests/ads.out:1:0
    expected:
    PASS
    feature_tests/ads.html:1333:7 The extension 'amp-analytics' was found on this page, but is unused (no 'amp-analytics' tag seen). This may become an error in the future. [DEPRECATION]
    saw:
    PASS
    feature_tests/ads.html:1334:7 The extension 'amp-analytics' was found on this page, but is unused (no 'amp-analytics' tag seen). This may become an error in the future. [DEPRECATION]
  Stack:
    AssertionError: 
    feature_tests/ads.out:1:0
    expected:
    PASS
    feature_tests/ads.html:1333:7 The extension 'amp-analytics' was found on this page, but is unused (no 'amp-analytics' tag seen). This may become an error in the future. [DEPRECATION]
    saw:
    PASS
    feature_tests/ads.html:1334:7 The extension 'amp-analytics' was found on this page, but is unused (no 'amp-analytics' tag seen). This may become an error in the future. [DEPRECATION]
        at ValidatorTestCase.run (/amphtml/validator/dist/validator_test_minified.js:948:398)
        at Object.<anonymous> (//amphtml/validator/dist/validator_test_minified.js:949:248)

3802 specs, 1 failure
Finished in 3.485 seconds
Traceback (most recent call last):
  File "build.py", line 580, in <module>
    Main()
  File "build.py", line 576, in Main
    RunTests(out_dir='dist', nodejs_cmd=nodejs_cmd)
  File "build.py", line 548, in RunTests
    subprocess.check_call([nodejs_cmd, '%s/test_runner' % out_dir])
  File "/usr/lib/python2.7/subprocess.py", line 540, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['node', 'dist/test_runner']' returned non-zero exit status 1
```

@rsimha-amp As to the errors we're expecting to see, I created a copy of the above mentioned PR in #11277. That should have the following error:

```
Failures:
1) ValidatorFeatures feature_tests/ads.html
  Message:
    AssertionError: 
    feature_tests/ads.out:1:0
    expected:
    PASS
    feature_tests/ads.html:1333:7 The extension 'amp-analytics' was found on this page, but is unused (no 'amp-analytics' tag seen). This may become an error in the future. [DEPRECATION]
    saw:
    PASS
    feature_tests/ads.html:1334:7 The extension 'amp-analytics' was found on this page, but is unused (no 'amp-analytics' tag seen). This may become an error in the future. [DEPRECATION]
  Stack:
    AssertionError: 
    feature_tests/ads.out:1:0
    expected:
    PASS
    feature_tests/ads.html:1333:7 The extension 'amp-analytics' was found on this page, but is unused (no 'amp-analytics' tag seen). This may become an error in the future. [DEPRECATION]
    saw:
    PASS
    feature_tests/ads.html:1334:7 The extension 'amp-analytics' was found on this page, but is unused (no 'amp-analytics' tag seen). This may become an error in the future. [DEPRECATION]
        at ValidatorTestCase.run (/amphtml/validator/dist/validator_test_minified.js:948:398)
        at Object.<anonymous> (//amphtml/validator/dist/validator_test_minified.js:949:248)

3802 specs, 1 failure
```

--- a/build-system/tasks/validator.js
+++ b/build-system/tasks/validator.js
@@ -16,21 +16,21 @@
 'use strict';
 
 var gulp = require('gulp-help')(require('gulp'));
-var execSync = require('child_process').execSync;
+var execOrDie = require('../exec.js').execOrDie;
 
 
 /**
  * Simple wrapper around the python based validator build.
  */
 function validator() {
-  execSync('cd validator && python build.py')
+  execOrDie('cd validator && python build.py')
 }
 
 /**
  * Simple wrapper around the python based validator webui build.
  */
 function validatorWebui() {
-  execSync('cd validator/webui && python build.py')
+  execOrDie('cd validator/webui && python build.py')
 }
 
 gulp.task('validator', 'Builds and tests the AMP validator.', validator);

